src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1979,14 +1979,35 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     // Arguments:
     // - initialUpdate: whether this font update should be considered as being
     //   concerned with initialization process. Value forwarded to event handler.
-    void TermControl::_UpdateFont(const bool initialUpdate)
+    winrt::fire_and_forget TermControl::_UpdateFont(const bool initialUpdate)
     {
         const int newDpi = static_cast<int>(static_cast<double>(USER_DEFAULT_SCREEN_DPI) * SwapChainPanel().CompositionScaleX());
 
         // TODO: MSFT:20895307 If the font doesn't exist, this doesn't
         //      actually fail. We need a way to gracefully fallback.
         _renderer->TriggerFontChange(newDpi, _desiredFont, _actualFont);
 
+        // If the actual font isn't what was requested...
+        if (_actualFont.GetFaceName() != _desiredFont.GetFaceName())
+        {
+            // Then warn the user that we picked something because we couldn't find their font.
+
+            // Save things we need to resume later.
+            auto strongThis{ get_strong() };
+
+            // Format message with user's choice of font and the font that was chosen instead.
+            const winrt::hstring message { fmt::format(std::wstring_view(RS_(L"NoticeFontNotFound")), _desiredFont.GetFaceName(), _actualFont.GetFaceName()) };

```

Comment:
Consider using `wstring_view{ }` instead of `()`. I'm trying to discourage the use of a constructor as a function call :smile: